bootstrap/terraform/vm.tf

```diff
@@ -0,0 +1,216 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "3.29.1"
+    }
+    azuread = {
+      source  = "hashicorp/azuread"
+      version = "2.30.0"
+    }
+    http = {
+      source  = "hashicorp/http"
+      version = "3.2.0"
+    }
+  }
+}
+
+data "azurerm_client_config" "current" {}
+provider "azuread" {}
+provider "azurerm" {
+  features {}
+}
+
+resource "azuread_application" "vault_azure_app" {
+  display_name = "vault_azure_app"
+}
+
+resource "azuread_service_principal" "vault_azure_sp" {
+  application_id = azuread_application.vault_azure_app.application_id
+}
+
+resource "azuread_service_principal_password" "vault_azure_sp_pwd" {
+  service_principal_id = azuread_service_principal.vault_azure_sp.id
+}
+
+# Use system assigned managed identity
+resource "azurerm_role_assignment" "vault_azure_msi_assignment" {
+  role_definition_name = "Reader"
+  scope                = azurerm_linux_virtual_machine.vault_azure_vm.id
+  principal_id         = azurerm_linux_virtual_machine.vault_azure_vm.identity[0].principal_id
+}
+
+# Uncomment below for user assigned managed identity
+# resource "azurerm_user_assigned_identity" "vault_azure_uid" {
+#   name                = "vault_azure_uid"
+#   resource_group_name = azurerm_resource_group.vault_azure_rg.name
+#   location            = azurerm_resource_group.vault_azure_rg.location
+# }
```

Comment:
Could/should we create resources for both identity types at the same time so contributors do not need to modify the Terraform project based on which method they want to setup?

Reply:
Yep, looks like we can create both (system-assigned and user-assigned) and let the user choose when getting the access token from IMDS. Done in [f2eb6a5](https://github.com/hashicorp/vault-plugin-auth-azure/pull/65/commits/f2eb6a53c924fdfce8e140e9e00c8c93e121cee1).